cg/meta/compress/compress.py

```diff
@@ -259,27 +264,38 @@ def update_fastq_hk(
             hk_fastq_first=hk_fastq_first, hk_fastq_second=hk_fastq_second
         )
 
-    def add_fastq_hk(self, sample_obj: Sample, fastq_first: Path, fastq_second: Path) -> None:
-        """Add FASTQ files to Housekeeper."""
-
-        if not sample_obj.application_version.application.is_external:
-            flow_cell_id: str = self.get_flow_cell_id(fastq_path=fastq_first)
-            fastq_tags: List[str] = [flow_cell_id, SequencingFileTag.FASTQ]
-        else:
-            fastq_tags: List[str] = [sample_obj.internal_id, SequencingFileTag.FASTQ]
+    def add_fastq_hk(
+        self,
+        spring_path: Path,
+        sample_internal_id: str,
+        fastq_first: Path,
+        fastq_second: Path,
+    ) -> None:
+        """Add decompressed FASTQ files to Housekeeper."""
+        spring_file: File = self.hk_api.files(path=spring_path.as_posix()).first()
+        fastq_tags = [
+            SequencingFileTag.FASTQ if tag_name == SequencingFileTag.SPRING else tag_name
+            for tag_name in [tag.name for tag in spring_file.tags]
+        ]
         LOG.info(
-            f"Adds {fastq_first}, {fastq_second} to bundle {sample_obj.internal_id} with tags {fastq_tags}"
+            f"Adds {fastq_first}, {fastq_second} to bundle {sample_internal_id} with tags {fastq_tags}"
         )
         if self.dry_run:
             return
 
         LOG.info("Updating files in Housekeeper...")
         for fastq in [fastq_first, fastq_second]:
             self.hk_api.add_and_include_file_to_latest_version(
-                bundle_name=sample_obj.internal_id, file=fastq, tags=fastq_tags
+                bundle_name=sample_internal_id, file=fastq, tags=fastq_tags
             )
         self.hk_api.commit()
 
+    @staticmethod
+    def get_flow_cell_id_from_sample(file_path: Path, sample: Sample) -> Optional[str]:
+        for flow_cell_name in [flow_cell.name for flow_cell in sample.flowcells]:
```

Comment:
Feel free to disregard. I would add this distinction so that we don't take a `Flowcell` for a `FlowCellDirectoryData`, as the name and id are switched between the two
```suggestion
        flow_cell: Flowcell
        for flow_cell_name in [flow_cell.name for flow_cell in sample.flowcells]:
```

Reply:
Removed method.